example-plugins/example_patient_sync/protocols/patient_sync.py

```diff
@@ -0,0 +1,155 @@
+        system_patient_id = self.lookup_external_id_by_system_url(canvas_patient, SYSTEM_URL)
+        update_patient_external_identifier = system_patient_id is None
+
+        # Here we check if the patient already has an external ID in Canvas for the partner platform
+        if not system_patient_id:
+            log.info(f">>> No external ID found for Canvas Patient ID {canvas_patient_id}:")
+
+            # Get the system external ID by making a GET request to the partner platform
+            system_patient = self.get_patient_from_system_api(canvas_patient_id)
+
+            system_patient_id = (
+                system_patient.json()["id"] if system_patient.status_code == 200 else None
+            )
+            log.info(
+                f">>>System patient ID for Canvas Patient ID {canvas_patient_id} is {system_patient_id}"
+            )
+            log.info(f">>> Need to update patient? {update_patient_external_identifier}")
+
+        # Great, now we know if the patient is assigned a system external ID with the partner
+        # platform, and if we need to update it. At this point the system_patient_id can be 3 values:
+        # 1. value we already had stored in Canvas,
+        # 2. value we just got from our GET API lookup, or
+        # 3. None
+        # And we have a true/false call to action: `update_patient_external_identifier`
+
+        # Generate the payload for creating or updating the patient in partner platform API
+        partner_payload = {
+            "externalId": canvas_patient.id,
+            "firstName": canvas_patient.first_name,
+            "lastName": canvas_patient.last_name,
+            "dateOfBirth": canvas_patient.birth_date.isoformat(),
+        }
+        # Add placeholder email when creating the patient if it's required
+        partner_payload["email"] = f"patient_{canvas_patient.id}@canvasmedical.com"
+        partner_payload["metadata"] = self.partner_patient_metadata
+
+        base_request_url = f"{self.partner_api_base_url}/patients/v2"
+
+        # If we HAVE a patient's partner external id, we know this is an update, so we'll append it to the request URL
+        request_url = (
+            f"{base_request_url}/{system_patient_id}" if system_patient_id else base_request_url
+        )
+
+        log.info(f">>> Partner platform API request URL: {request_url}")
+        log.info(f">>> Partner platform API patient payload: json={partner_payload}")
+        log.info(
+            f">>> Partner platform API request headers: headers={self.partner_request_headers}"
+        )
+
+        resp = http.post(request_url, json=partner_payload, headers=self.partner_request_headers)
+        log.info(f">>> Partner platform API response: status_code={resp.status_code}, json={resp.json()}")
+
+        # If the request was successful, we should now have a system patient ID if we didn't before
+        if system_patient_id is None:
+            system_patient_id = resp.json().get("id")
```

Comment:
I would add something like "If your system's API returns the ID of the newly created record, grab it from the response so we can add it to the Canvas patient record."